CMakeLists.txt

```diff
@@ -19,6 +19,7 @@ set(VERSION_MAJOR 1)
 set(VERSION_MINOR 0)
 set(VERSION_PATCH 0)
 
+option(BUILD_TESTING "Turn BUILD_TESTING=on by default" ON)
```

Comment:
When documenting these options, it's good to try and think about this from the perspective of some user new to our library. 

In it's current form, the definition is a little bit recursive, and doesn't really tell you what `BUILD_TESTING` _does_.
```suggestion
option(BUILD_TESTING "Build tests for s2n-tls. By default only unit tests are built." ON)
```